packages/app-store/salesforce/lib/CrmService.ts

```diff
@@ -711,4 +717,106 @@ export default class SalesforceCRMService implements CRM {
+    bookingUid?: string
+  ) {
+    const conn = await this.conn;
+    const appOptions = this.getAppOptions();
+    const { createEventOn, onBookingWriteToRecordFields } = appOptions;
+
+    // Figure out what we're writing to
+    let personRecordType = appOptions.createEventOn;
+
+    if (
+      createEventOn === SalesforceRecordEnum.LEAD &&
+      appOptions.createEventOnLeadCheckForContact &&
+      this.getFallbackToContact()
+    ) {
+      personRecordType = SalesforceRecordEnum.CONTACT;
+    }
+
+    // Search the fields and ensure 1. they exist 2. they're the right type
+    const fieldsToWriteOn = Object.keys(onBookingWriteToRecordFields);
+
+    const existingFields = await this.ensureFieldsExistOnObject(fieldsToWriteOn, personRecordType);
+
+    const existingFieldNames = existingFields.map((field) => field.name);
+
+    const personRecordQuery = await conn.query(
+      `SELECT ${existingFieldNames.join(", ")} FROM ${personRecordType} WHERE Id = '${contactId}'`
+    );
+
+    if (!personRecordQuery.records.length) {
+      this.log.warn(`Could not write to person record id ${contactId}`);
+      return;
+    }
+
+    const personRecord = personRecordQuery.records[0] as { [key: string]: any };
+
+    const writeOnRecordBody: { [key: string]: any } = {};
+
+    for (const salesforceField of existingFields) {
+      const fieldToWriteOn = onBookingWriteToRecordFields[salesforceField.name];
+
+      // Determine if we write to the field
+      if (
+        fieldToWriteOn.whenToWrite === WhenToWriteToRecord.FIELD_EMPTY &&
+        !!personRecord[salesforceField.name]
+      )
+        continue;
+
+      // Determine if the field to write on matches the field type
+      if (fieldToWriteOn.fieldType === salesforceField.type) {
+        if (salesforceField.type === SalesforceFieldType.TEXT) {
+          const stringToWrite = fieldToWriteOn.value.substring(0, salesforceField.length);
+          writeOnRecordBody[salesforceField.name] = stringToWrite;
+        }
+
+        if (salesforceField.type === SalesforceFieldType.DATE) {
+          if (fieldToWriteOn.value === DateFieldTypeData.BOOKING_START_DATE) {
+            writeOnRecordBody[salesforceField.name] = new Date(startTime).toISOString();
+          } else if (fieldToWriteOn.value === DateFieldTypeData.BOOKING_CREATED_DATE) {
+            // Determine the createdAt from the event
+            if (!bookingUid) {
+              this.log.warn(`No uid for booking with organizer ${organizerEmail}`);
+            }
+
+            const booking = await prisma.booking.findFirst({
+              where: {
+                uid: bookingUid,
+              },
+              select: {
+                createdAt: true,
+              },
+            });
+
+            if (!booking) {
+              this.log.warn(`No booking found for ${bookingUid}`);
+              return;
+            }
+
+            writeOnRecordBody[salesforceField.name] = new Date(booking.createdAt).toISOString();
```

Comment:
booking.creaedAt is already in ISO format

Reply:
What if we left it as is? Right now it's typed as a `Date` but when calling `.toISOString()` it converts it to type `string` which the function `getDateFieldValue` expects as a return value.